invoice/src/main/java/org/killbill/billing/invoice/usage/ContiguousIntervalUsageInArrear.java

```diff
@@ -426,12 +437,12 @@ BigDecimal computeToBeBilledConsumableInArrearWith_ALL_TIERS(final List<TieredBl
                 nbUsedTierBlocks = tmp;
                 remainingUnits = 0;
             }
-            result = result.add(tieredBlock.getPrice().getPrice(getCurrency()).multiply(new BigDecimal(nbUsedTierBlocks)));
+            toBeBilledDetails.add(new ToBeBilledConsumableInArrearDetail(tierNum, tieredBlock.getUnit().getName(), tieredBlock.getPrice().getPrice(getCurrency()), nbUsedTierBlocks));
         }
-        return result;
+        return toBeBilledDetails;
     }
 
-    BigDecimal computeToBeBilledConsumableInArrearWith_TOP_TIER(final List<TieredBlock> tieredBlocks, final Long units) throws CatalogApiException {
+    ToBeBilledConsumableInArrearDetail computeToBeBilledConsumableInArrearWith_TOP_TIER(final List<TieredBlock> tieredBlocks, final Long units, int tierNum) throws CatalogApiException {
```

Comment:
Same remark as `UsageType.CAPACITY`: Should we also add the details for `UsageType.CONSUMABLE` and `TOP_TIER`: This is simpler since there will be only one item, but writing the detail about the chosen tier might be useful.